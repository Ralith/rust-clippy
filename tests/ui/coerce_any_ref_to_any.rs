#![warn(clippy::coerce_any_ref_to_any)]

use std::any::Any;

fn main() {
    let x: Box<dyn Any> = Box::new(());
    let ref_x = &x;

    f(&x);
    //~^ coerce_any_ref_to_any

    f(ref_x);
    //~^ coerce_any_ref_to_any

    let _: &dyn Any = &x;
    //~^ coerce_any_ref_to_any

    f(&42);
    f(&Box::new(()));
    f(&**ref_x);
    f(&*x);
    let _: &dyn Any = &*x;
}

fn f(_: &dyn Any) {}
