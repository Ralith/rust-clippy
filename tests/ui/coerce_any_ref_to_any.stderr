error: coercing `&std::boxed::Box<dyn std::any::Any>` to `&dyn Any` rather than dereferencing to the `dyn Any` inside
  --> tests/ui/coerce_any_ref_to_any.rs:9:7
   |
LL |     f(&x);
   |       ^^ help: consider dereferencing: `&*x`
   |
   = note: `-D clippy::coerce-any-ref-to-any` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::coerce_any_ref_to_any)]`

error: coercing `&std::boxed::Box<dyn std::any::Any>` to `&dyn Any` rather than dereferencing to the `dyn Any` inside
  --> tests/ui/coerce_any_ref_to_any.rs:12:7
   |
LL |     f(ref_x);
   |       ^^^^^ help: consider dereferencing: `&**ref_x`

error: coercing `&std::boxed::Box<dyn std::any::Any>` to `&dyn Any` rather than dereferencing to the `dyn Any` inside
  --> tests/ui/coerce_any_ref_to_any.rs:15:23
   |
LL |     let _: &dyn Any = &x;
   |                       ^^ help: consider dereferencing: `&*x`

error: aborting due to 3 previous errors

